lighthouse/src/main.rs

```diff
@@ -525,33 +532,7 @@ fn run<E: EthSpec>(
     let logfile_restricted = !matches.get_flag("logfile-no-restricted-perms");
 
     // Construct the path to the log file.
-    let mut log_path: Option<PathBuf> = clap_utils::parse_optional(matches, "logfile")?;
-    if log_path.is_none() {
-        log_path = match matches.subcommand() {
-            Some(("beacon_node", _)) => Some(
-                parse_path_or_default(matches, "datadir")?
-                    .join(DEFAULT_BEACON_NODE_DIR)
-                    .join("logs")
-                    .join("beacon")
-                    .with_extension("log"),
-            ),
-            Some(("validator_client", vc_matches)) => {
-                let base_path = if vc_matches.contains_id("validators-dir") {
-                    parse_path_or_default(vc_matches, "validators-dir")?
-                } else {
-                    parse_path_or_default(matches, "datadir")?.join(DEFAULT_VALIDATOR_DIR)
-                };
-
-                Some(
-                    base_path
-                        .join("logs")
-                        .join("validator")
-                        .with_extension("log"),
-                )
-            }
-            _ => None,
-        };
-    }
+    let log_path: Option<PathBuf> = clap_utils::parse_optional(matches, "logfile")?;
```

Comment:
We actually need to retain the check here.
If the subcommand is either `beacon_node` or `validator_client` AND the logfile flag wasn't supplied, log to the default location of `$DATADIR/logs/$service.log`. Should be fine to copy most of the deleted code to restore this functionality.